python/ray/tests/test_autoscaler.py

```diff
@@ -35,6 +36,47 @@
 import pytest
 
 
+class MockNodeInfoStub():
+    """Mock for GCS node info stub used by autoscaler to drain Ray nodes.
+    """
+
+    def DrainNode(self, drain_node_request):
+        """Return a response indicating successful drain to the autoscaler."""
+        node_ids_to_drain = [
+            data_item.node_id
+            for data_item in drain_node_request.drain_node_data
+        ]
+
+        ok_gcs_status = gcs_service_pb2.GcsStatus(
+            code=0, message="Yeah, it's fine.")
+
+        all_nodes_drained_status = gcs_service_pb2.DrainNodeStatus(
+            status=ok_gcs_status,
+            drain_node_status=[
+                gcs_service_pb2.DrainNodeStatus(node_id=node_id)
+                for node_id in node_ids_to_drain
+            ])
+
+        return gcs_service_pb2.DrainNodeReply(
+            drain_node_status=all_nodes_drained_status)
+
+
+def mock_raylet_id() -> bytes:
+    """Random raylet id to pass to load_metrics.update.
+    """
+    return os.urandom(10)
+
+
+def fill_in_raylet_ids(provider, load_metrics) -> None:
+    """Raylet ids for each ip are usually obtained by polling the GCS
+    in monitor.py. For test purposes, we sometimes need to manually fill
+    these fields with mocks.
+    """
+    for node in provider.non_terminated_nodes({}):
+        ip = provider.internal_ip(node)
+        load_metrics.raylet_id_by_ip[ip] = mock_raylet_id()
```

Comment:
Added this function and put it all over the place to get tests to pass.
Then changed autoscaler so that we do not attempt to drain nodes whose ips are not in load metrics, which mostly removes the need for this function..
Will just leave it as is, as it makes the tests slightly more "realistic" (raylet_ids are present in load metrics, as they usually would be in real life.)